File: 367-valid-perfect-square.py
# return True if valid perfect square, False if not

# still don't fully understand the math of this... look into more maybe

class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        num1 = 0
        num2 = 1

        while num1 < num:
            num1 += num2
            num2 += 2

        return num1 == num

File: 367-valid-perfect-square.js
// look at py file for problem

var isPerfectSquare = function(num) {
    let num1 = 0;
    let num2 = 1;

    while (num1 < num) {
        num1 += num2;
        num2 += 2;
    }

    return num1 === num;
};
